src/IO/S3Common.cpp

```diff
@@ -135,9 +135,15 @@ void AuthSettings::updateFrom(const AuthSettings & from)
     headers = from.headers;
     region = from.region;
     server_side_encryption_customer_key_base64 = from.server_side_encryption_customer_key_base64;
-    use_environment_credentials = from.use_environment_credentials;
-    use_insecure_imds_request = from.use_insecure_imds_request;
-    expiration_window_seconds = from.expiration_window_seconds;
+
+    if (from.use_environment_credentials)
+        use_environment_credentials = from.use_environment_credentials;
```

Comment:
does this check (`if (from.use_environment_credentials)`) make sense? for example use_environment_credentials was true, then we wanted to change it to false and want this setting to apply without server restart.

Reply:
yes because it's `std::optional` but it's not explicit, will add `has_value`.

Reply:
ah, did not notice that is is std::optional

Reply:
should be more obvious now 😄